### PR TITLE
Implement API for creating, updating and deleting refs.

### PR DESCRIPTION
**Pull request by imeszaros**
### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [x] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [x] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

This pull requests implements create, update and delete operations on refs, according to GitHub API specs.

I tried to follow existing conventions. Did not implement "get all refs" since the existing get API already implements it if I understood it correctly.

I'm not a Scala pro so i'm sure there is room for improvements.. :)

In the first commit I also fixed a bug I discovered during testing my changes.

### CODE REVIEW

File: src/main/scala/gitbucket/core/controller/api/ApiIssueCommentControllerBase.scala [hunk 1]
```diff
@@ -85,7 +85,7 @@ trait ApiIssueCommentControllerBase extends ControllerBase {
    * iv. Delete a comment
    * https://docs.github.com/en/rest/reference/issues#delete-an-issue-comment
    */
-  delete("/api/v3/repos/{owner}/{repo}/issues/comments/:id")(readableUsersOnly { repository =>
+  delete("/api/v3/repos/:owner/:repo/issues/comments/:id")(readableUsersOnly { repository =>
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/controller/api/ApiIssueCommentControllerBase.scala:88]: Oh, it wasn't working...😲

Comment (onukura) [reply src/main/scala/gitbucket/core/controller/api/ApiIssueCommentControllerBase.scala:88]: I didn't notice this until now...!

File: src/main/scala/gitbucket/core/api/CreateARef.scala [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package gitbucket.core.api
+
+/**
+ * https://developer.github.com/v3/git/refs/#create-a-reference
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/api/CreateARef.scala:4]: Links to GitHub docs should point to https://docs.github.com/. Though, there are still many old links in GitBucket codebase, we are updating these links when we touch code around them one by one. Could you update some links which is covered by this pull request?

**Review by takezoe — APPROVED** [review]
@imeszaros Sorry for my late review. This looks good to me other than links to GitHub docs in comments.

File: src/main/scala/gitbucket/core/api/CreateARef.scala (hunk 2 above)
Comment (imeszaros) [reply src/main/scala/gitbucket/core/api/CreateARef.scala]: Yeah no problem, I fixed the links.

**Comment by takezoe** [comment]
Thanks for updating links!

**Comment by takezoe** [comment]
I new endpoints added by this pull request to Wiki page.
https://github.com/gitbucket/gitbucket/wiki/API-WebHook

**Comment by takezoe** [comment]
> I tried to follow existing conventions. Did not implement "get all refs" since the existing get API already implements it if I understood it correctly.

By the way, the current GitHub API seems to not have "get all refs" API. Instead it has "list matching references" API.
https://docs.github.com/en/free-pro-team@latest/rest/reference/git#list-matching-references

**Pull request merged**